### PR TITLE
test: ctypes and relative include paths

### PR DESCRIPTION
**Pull request by emillon**
This adds a test that shows that under `(using ctypes 0.2)`, local headers need to be added in two different locations because rules are executed from different places. See #5325.

### CODE REVIEW

File: test/blackbox-tests/test-cases/ctypes/directories.t [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  >    (functor f)
+  >    (instance functions))
+  >   (generated_entry_point entry)))
+  > EOF
+  > }
+
+  $ mk_dune "-I lib -I ." > lib/dune
+
+  $ cat > lib/lib.h << EOF
+  > #include <fcntl.h>
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/ctypes/directories.t:46]: Can we use a header that would be testable on windows? At least in theory. Something that's part of the C standard would be ideal.

File: test/blackbox-tests/test-cases/ctypes/directories.t [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  >   let open_ = foreign "open" (ptr char @-> int @-> returning int)
+  > end
+  > EOF
+
+  $ dune build
+
+We ensure that just `-I lib` or `-I .` are not enough on their own.
+
+  $ mk_dune "-I lib" > lib/dune
+  $ dune build > /dev/null 2>&1
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/ctypes/directories.t:72]: Anyway to get a reproducible chunk of the error?

File: test/blackbox-tests/test-cases/ctypes/directories.t (hunk 1 above)
Comment (emillon) [reply test/blackbox-tests/test-cases/ctypes/directories.t:46]: Ah, I was looking for something in the libc but forgot that it's not that standard. I'll try with something from stdlib.h or string.h.

File: test/blackbox-tests/test-cases/ctypes/directories.t (hunk 2 above)
Comment (emillon) [reply test/blackbox-tests/test-cases/ctypes/directories.t:72]: I'm not sure what's best - `gcc` exits with an error in both cases. I was thinking we could look for `cc` in the verbose log and cut where the command is executed (in one case it's from `_build/default` directly) but that seems even more fragile.

File: test/blackbox-tests/test-cases/ctypes/directories.t (hunk 1 above)
Comment (emillon) [reply test/blackbox-tests/test-cases/ctypes/directories.t]: done using `BUFSIZ` and `fopen`.

**Review by rgrinberg — APPROVED** [review]

**Pull request merged**